### ISSUE TITLE
I get different test accuracy :)

### ISSUE THREAD

**Issue by woolfel** [ElephantHunters/Fashion-MNIST-WITH-Deep-Learning-Studio#1]
I came across your article about fashion mnist over the weekend and decided to try your configuration. At the end of 100 epocs I get test_loss=0.2078 and test_acc=0.9232. Your article states you got 95% accuracy. Curious what kind of hardware are you using?
My workstation has NVidia 1070, Cuda 9, 64G Ram, Tensorflow 1.10 and latest keras. I've tried a couple of different configurations that gets 93% test accuracy with a lot fewer epocs and smaller batches. 

# TensorFlow and tf.keras
import tensorflow as tf
from tensorflow import keras

# Helper libraries
import numpy as np
import matplotlib.pyplot as plt

print(tf.__version__)

batch_size=20
num_classes = 10
epochs = 20

# input image dimensions
img_rows, img_cols = 28, 28

# the data, split between train and test sets
fashion_mnist = keras.datasets.fashion_mnist
(x_train, y_train), (x_test, y_test) = fashion_mnist.load_data()

# Create instance of DataGenerator and use shift, h flip and zoom
datagen = keras.preprocessing.image.ImageDataGenerator(width_shift_range=.118,height_shift_range=.118,horizontal_flip=True,zoom_range=.021)

x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
input_shape = (img_rows, img_cols, 1)

x_train = x_train.astype('float32')
x_test = x_test.astype('float32')
x_train /= 255
x_test /= 255
print('x_train shape:', x_train.shape)
print(x_train.shape[0], 'train samples')
print(x_test.shape[0], 'test samples')

# convert class vectors to binary class matrices
y_train = keras.utils.to_categorical(y_train, num_classes)
y_test = keras.utils.to_categorical(y_test, num_classes)

# Setup the Keras model
model = keras.Sequential()
model.add(keras.layers.Conv2D(64, kernel_size=(2, 2),
                 activation='relu',
                 input_shape=input_shape))
model.add(keras.layers.Conv2D(64, (2, 2), activation='relu'))
model.add(keras.layers.MaxPooling2D(pool_size=(2, 2)))
model.add(keras.layers.Dropout(0.1589))
model.add(keras.layers.Flatten())
model.add(keras.layers.Dense(128, activation='relu'))
model.add(keras.layers.Dropout(0.5687))
model.add(keras.layers.Dense(num_classes, activation='softmax'))

# Compile the model
model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=tf.train.AdamOptimizer(),
              metrics=['accuracy'])

datagen.fit(x_train)
# calling fit_geneator will use the DataGenerator to modify the images in real-time
model.fit_generator(datagen.flow(x_train, y_train, batch_size=batch_size),
                    steps_per_epoch=len(x_train) / batch_size, epochs=epochs, validation_data=(x_test, y_test))

# Retrain on the original image set to increase test accuracy
model.fit(x_train, y_train,
          batch_size=batch_size,
          epochs=10,
          verbose=1,
          validation_data=(x_test, y_test))
                                                            
score = model.evaluate(x_test, y_test, verbose=0)
print('Test loss:', score[0])
print('Test accuracy:', score[1])

model.save('./saved_model.h5')
